components/server/src/github/github-auth-provider.ts

```diff
@@ -98,13 +98,24 @@ export class GitHubAuthProvider extends GenericAuthProvider {
                 .map((s: string) => s.trim())
             );
 
+            const filterPrimaryEmail = (emails: GitHub.UsersListEmailsResponse) => {
+                // if there is any verified email with a domain that is in the blockNewUsersPassList then use this email as primary email
+                const emailDomainInPasslist = (mail: string) => this.env.blockNewUsersPassList.some(e => mail.endsWith(`@${e}`));
+                const result = emails.filter(e => e.verified).filter(e => emailDomainInPasslist(e.email))
+                if (result.length > 0) {
+                    return result[0].email;
+                }
```

Comment:
Maybe move this entire block into an `if (this.env.blockNewUsers) { }` block? (To avoid confusing side-effects if you have a `blockNewUsersPassList` but `blockNewUsers` is false)

Reply:
Done :+1: